exec/java-exec/src/main/java/org/apache/calcite/jdbc/DynamicRootSchema.java

```diff
@@ -78,7 +78,7 @@ public void loadSchemaFactory(String schemaName, boolean caseSensitive) {
     try {
       SchemaPlus schemaPlus = this.plus();
       StoragePlugin plugin = getSchemaFactories().getPlugin(schemaName);
-      if (plugin != null) {
+      if (plugin != null && plugin.getConfig().isEnabled()) {
```

Comment:
Well, generally I would say the exception describes what actually is. Schema is not valid, it means it does not exist or disabled. I am not sure we should change anything. Plus unclear where to throw it. Definitely not in this method.

Reply:
It appears not easy to add this new exception. We should keep disabled plugins list and their schemas and then in the process of validation it is necessary to find the actual schema in this list. But since the new message is questionable, let's leave it as is.